### PR TITLE
feat(payments-next): add processing page

### PR DESCRIPTION
**Pull request by StaberindeZA**
## Because

- After subscribe now button is pressed or paypal payment is completed the customer should be navigated to a processing page.

## This pull request

- Adds the processing page.
- Adds a Loading Spinner for payments-next

## Issue that this pull request solves

Closes: #FXA-7577

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/processing/page.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  const locale = headers().get('accept-language') || DEFAULT_LOCALE;
+  const l10n = getApp().getL10n(locale);
+
+  return (
+    <section className="flex flex-col" data-testid="payment-processing">
+      <div className="flex flex-col text-center text-sm">
+        <LoadingSpinner l10n={l10n} />
+        {l10n.getString(
+          'payment-processing-message',
+          'Please wait while we process your payment...'
```
Comment (xlisachan) [inline apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/processing/page.tsx:15]: I think the `...` at the end has to be `…` similar to how it is in the ftl id.

File: apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/processing/page.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import { DEFAULT_LOCALE } from '@fxa/shared/l10n';
+import { headers } from 'next/headers';
+
+export default async function ProcessingPage() {
+  const locale = headers().get('accept-language') || DEFAULT_LOCALE;
+  const l10n = getApp().getL10n(locale);
+
+  return (
+    <section className="flex flex-col" data-testid="payment-processing">
+      <div className="flex flex-col text-center text-sm">
```
Comment (xlisachan) [inline apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/processing/page.tsx:11]: Could we eliminate the nested div by making the section's classname `flex flex-col text-center text-sm` instead?

File: apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/processing/page.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { getApp, LoadingSpinner } from '@fxa/payments/ui/server';
```
Comment (xlisachan) [inline apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/processing/page.tsx:1]: :eyes:
```
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 ```

**Review by xlisachan — APPROVED** [review]
Just some minor updates, otherwise LGTM! Thanks Reino!

**Review by bcolsson — APPROVED** [review]

File: apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/processing/page.tsx (hunk 2 above)
Comment (StaberindeZA) [reply apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/processing/page.tsx:11]: Ah yes much better. Thank you.

**Pull request merged**